promtail: enable structured_metadata convert labels (#10752)
grafana/loki: docs/sources/send-data/promtail/stages/structured_metadata.md

```

The first stage would extract `stream` with a value of `stderr` and `traceID` with a value of `0242ac120002` into
the extracted data set. The `labels` stage would turn that `stream` and `stderr` key-value pair into a stream label.
The `structured_metadata` stage would attach the `traceID` and `0242ac120002` key-value pair as a structured metadata to the log line.

### Parse from service discovery labels

For the configuration below, you can use labels from `relabel_configs` to use as `structured_metadata`:

```yaml
pipeline_stages:
  - structured_metadata:
      pod_uid:
      pod_host_ip:
relabel_configs:
  - action: replace
    source_labels:
      - __meta_kubernetes_pod_uid
    target_label: pod_uid
  - action: replace
    source_labels:
      - __meta_kubernetes_pod_host_ip
    target_label: pod_host_ip
```

Given the following discovered labels below with a log line `sample log`:

|- Discovered label |- Value |- Target label |
| - | - | - |
| __meta_kubernetes_pod_host_ip | 127.0.0.1 | pod_host_ip |
| __meta_kubernetes_pod_uid | b3937321-fe90-4e15-ac94-495c8fdb9202 | pod_uid |

> The `structured_metadata` stage would turn the discovered labels `pod_uid` and `pod_host_ip` in key-value pair as a structured metadata to the log line `sample log`
> excluding them out of creating high-cardinality streams.

Reviewer: ```suggestion
The `structured_metadata` stage would turn the discovered labels `pod_uid` and `pod_host_ip` into key-value pairs as  structured metadata for the log line `sample log` and
exclude them from creating high-cardinality streams.
```
